model.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+"""model_project.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/19prXM_4R54HKfpar8kqpqUKQD8wcF5w-
+"""
+
+import pandas as pd
+from sklearn.model_selection import train_test_split
+import numpy as np
+from sklearn.dummy import DummyClassifier
+import seaborn as sns
+import matplotlib.pyplot as plt
+from sklearn.preprocessing import StandardScaler
+from sklearn.decomposition import PCA
+from sklearn.linear_model import LogisticRegression
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.pipeline import Pipeline
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+from sklearn import svm
+from imblearn.over_sampling import SMOTE
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.ensemble import RandomForestClassifier, AdaBoostClassifier
+import io
+
+from google.colab import files
+uploaded = files.upload()
+
+#import train dataset
+X = pd.read_csv(io.BytesIO(uploaded['train_data.csv']))
+
+#import labels
+y = pd.read_csv(io.BytesIO(uploaded['train_labels.csv']))
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+#y = y['y'].apply(lambda x: 1 if x == -1 else 0)
+
+
+#baseline
+strategies = ['stratified', 'most_frequent', 'prior', 'uniform']
+
+for i in strategies:
+    dummy_clf = DummyClassifier(strategy=i)
+    dummy_clf.fit(X_train, y_train)
+    score = dummy_clf.score(X_train, y_train)
+    dum_labels = np.unique(dummy_clf.predict(X))
+    print(f'For {i} strategy score is {score}')
+    plt.bar(i, score)
+
+
+#scaling data
+
+scaler = StandardScaler()
+
+X_train_scale = scaler.fit_transform(X_train)
+y_train_scale = scaler.fit_transform(y_train)
+
+
+#PCA
+
+pca = PCA(n_components=2, whiten=True)
+X_pca = pca.fit_transform(X_train_scale)
+
+
+#Logistic Regression
+
+clf = LogisticRegression(random_state=0).fit(X, y)
+
+#KNeighborsClassifier
+
+knn = KNeighborsClassifier().fit(X, y)
```

Comment:
Short function names are useful, but may not be understood by the recipient. This is worth thinking about.